cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/notifier/SelfHealingNotifier.java

```diff
@@ -152,7 +152,8 @@ private void updateSelfHealingStateChange(AnomalyType anomalyType, Boolean oldVa
       long enabledTimeMs = _selfHealingEnabledHistoricalDurationMs.get(anomalyType);
       if (_selfHealingEnabled.get(anomalyType)) {
         // Add current duration during which the self-healing is enabled.
-        enabledTimeMs += nowMs - _selfHealingStateChangeTimeMs.get(false).get(anomalyType);
+        Long currentEnabledSelfHealingStartTime = _selfHealingStateChangeTimeMs.get(false).get(anomalyType);
+        enabledTimeMs += nowMs - (currentEnabledSelfHealingStartTime == null ? 0L : currentEnabledSelfHealingStartTime);
```

Comment:
I think here should be `_notifierStartTimeMs` instead of `0L`.
`currentEnabledSelfHealingStartTime == null` happens only when CC is running with self-healing on by default, so the base time to calculate the time self-healing  being on should be CC start time(i.e. `_notifierStartTimeMs`).

Reply:
Correct -- nice catch!